server/src/main/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorService.java

```diff
@@ -252,9 +259,16 @@ public SimpleHealthIndicatorDetails getDetails() {
 
     }
 
-    private static String getTruncatedIndicesString(Set<String> indices) {
+    private static String getTruncatedIndicesString(Set<String> indices, Metadata clusterMetadata) {
         final int maxIndices = 10;
-        String truncatedIndicesString = indices.stream().limit(maxIndices).collect(joining(", "));
+        Comparator<String> comparePrioritiesAscendingComparator = Comparator.comparingInt(index -> clusterMetadata.index(index).priority());
```

Comment:
nit: reversed could be called here to avoid `comparePrioritiesAscendingComparator` coming to life altogether (the jit will likely not allocate it anyway so it's more of a readbility suggestion)

Reply:
Oops. That was an easy fix!